cmd/serverless/main.go

```diff
@@ -209,6 +209,11 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 	if _, err := config.Load(); err != nil {
 		log.Errorf("Error happened when loading configuration from datadog.yaml for metric agent: %s", err)
 	}
+
+	if err := sendAPIKeyToShell(os.Getenv(apiKeyEnvVar)); err != nil {
```

Comment:
One thing I'm not sure at all about: is the environment variable set if the key is read from a config file ?
If not you probably want to use `config.Datadog.GetString("api_key")`.

Reply:
no we set it via env a bit earlier in the code after KMS decryption for instance